include/irr/asset/ICPUPipelineLayout.h

```diff
@@ -58,11 +58,25 @@ class ICPUPipelineLayout : public IAsset, public IPipelineLayout<ICPUDescriptorS
 			    m_pushConstantRanges = nullptr;
 		}
 
-        bool canBeRestoredFrom(const IAsset* _other) const override
+        bool canBeRestoredFrom_recurseDAG(const IAsset* _other) const override
         {
-            if (!IAsset::canBeRestoredFrom(_other))
-                return false;
+            auto* other = static_cast<const ICPUPipelineLayout*>(_other);
 
+            if ((!m_pushConstantRanges || !other->m_pushConstantRanges) && m_pushConstantRanges != other->m_pushConstantRanges)
+                return false;
+            if (m_pushConstantRanges->size() != other->m_pushConstantRanges->size())
+                return false;
```

Comment:
this is bad, its okay to have no push constant ranges, this will fail if either is 0